In [1]:
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [2]:
test_df = pd.read_csv('Doceree-HCP_Test.csv', encoding='latin1')
train_df = pd.read_csv('Doceree-HCP_Train.csv', encoding='latin1')
#SAMPLE_df = pd.read_csv('Doceree_HCP_Sample_Submission_1.csv')

In [3]:
test_df.head(10)

ID DEVICETYPE  PLATFORM_ID     BIDREQUESTIP  \
0  115501    Desktop            2   75.189.231.103   
1  115502     Mobile            2    24.101.33.158   
2  115503    Desktop            2  172.118.216.142   
3  115504    Desktop            7   71.105.120.171   
4  115505    Desktop            2     73.82.211.73   
5  115506    Desktop            2  169.147.254.129   
6  115507    Desktop            9      66.249.66.2   
7  115508    Desktop            2   50.236.245.242   
8  115509    Desktop            2    204.99.118.10   
9  115510     Mobile            2     71.228.67.64   

                        USERPLATFORMUID       USERCITY  USERZIPCODE  \
0  0d5041ff-f0b6-4d1a-9ad7-0a29f7d485b4   Fayetteville      28305.0   
1  c8396dd0-969f-4d99-a40b-b7bb1f516154  Conneaut Lake      16316.0   
2  3c97a081-6518-43f8-9f26-369759cfb471         Covina      91724.0   
3  3e2578c8-f794-41af-a38c-c5cfb3c0f014       Brooklyn      11226.0   
4  ec2ae7ce-6a8c-4156-98a7-07203e60f483       Marietta      30062.0   
5  0313febc-12e6-4289-8d74-5f0c1aecc385    Kansas City      64113.0   
6  ce1f2f66-056e-4684-8c24-6d70e632a269            NaN          NaN   
7  8fc0a64c-0c71-4888-aff4-833823802bba        Seattle      98188.0   
8  12ec92c0-a55b-45a2-8c4e-4237dcfc8eef     Scottsdale      85260.0   
9  d0a2d8ab-6ea1-440e-a2f1-03acd6a1db99       Sarasota      34232.0   

                                           USERAGENT            PLATFORMTYPE  \
0  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6...  Online Medical Journal   
1  Mozilla/5.0 (iPhone; CPU iPhone OS 15_6_1 like...  Online Medical Journal   
2  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7...  Online Medical Journal   
3  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7...  Online Medical Journal   
4  Mozilla/5.0 (Windows NT 10.0; Win64; x64) Appl...  Online Medical Journal   
5  Mozilla/5.0 (Windows NT 10.0; Win64; x64) Appl...  Online Medical Journal   
6  Mozilla/5.0 (Linux; Android 6.0.1; Nexus 5X Bu...  Online Medical Journal   
7  Mozilla/5.0 (Windows NT 10.0; Win64; x64) Appl...  Online Medical Journal   
8  Mozilla/5.0 (Windows NT 10.0; Win64; x64) Appl...  Online Medical Journal   
9  Mozilla/5.0 (iPhone; CPU iPhone OS 16_3_1 like...  Online Medical Journal   

  CHANNELTYPE                                                URL  \
0     Website  https://www.clinicaladvisor.com/home/features/...   
1     Website  https://www.ophthalmologyadvisor.com/topics/ca...   
2     Website  https://www.psychiatryadvisor.com/author/tori-...   
3     Website  https://www.cureus.com/articles/105482-diverti...   
4     Website  https://www.renalandurologynews.com/home/confe...   
5     Website  https://www.renalandurologynews.com/home/decis...   
6     Website  https://www.2minutemedicine.com/studygraphics-...   
7     Website  https://www.cancertherapyadvisor.com/home/canc...   
8     Website                 https://www.empr.com/drug/evrysdi/   
9     Website  https://www.clinicaladvisor.com/home/dermatolo...   

                                            KEYWORDS  
0  Family Practice|Drainage|Clinical|Dermatology|...  
1  General|Clinical|Operative|Medicine|Cardiology...  
2  Abortion|Anxiety Disorders|Apnea|False|Trauma|...  
3  Health|Male|Neurological Surgery|Otolaryngolog...  
4  chronic kidney disease|pain|nephrology|disease...  
5  Abscess|Liver Transplantation|Transplantation|...  
6                             the|medicine|oncology|  
7  General|Endocrine|False|Medicine|Surgery|Urolo...  
8  Medicine|Metabolic|Dermatology|Child|False|Mal...  
9  Hemoptysis|Oral|Rheumatology|Cardiovascular|Sm...

In [4]:
train_df.head()

ID DEVICETYPE  PLATFORM_ID    BIDREQUESTIP  \
0  1001    Desktop            2    170.173.0.22   
1  1002    Desktop            2   65.216.253.25   
2  1003    Desktop            2    66.232.79.22   
3  1004    Desktop            3  137.54.125.246   
4  1005     Mobile            7  174.202.231.99   

                        USERPLATFORMUID     USERCITY  USERZIPCODE  \
0  6974dcaa-f932-480e-9fb5-c52e20e1393a     Portland      97206.0   
1  c12f3f8f-8fcf-484a-90e1-1ac04db8cdcf    Arlington      22202.0   
2  a698de4b-e200-46dd-b5fb-40402175ae18  New Meadows      83654.0   
3  45967533-75c8-4fbd-a00c-e6ff20447aaa          NaN  229114624.0   
4  a17e25be-532d-4cf5-b916-9308c8c3961f      Houston      77008.0   

                                           USERAGENT            PLATFORMTYPE  \
0  Mozilla/5.0 (Windows NT 10.0; Win64; x64) Appl...  Online Medical Journal   
1  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7...  Online Medical Journal   
2  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7...  Online Medical Journal   
3  Mozilla/5.0 (Windows NT 10.0; Win64; x64) Appl...  Online Medical Journal   
4  Mozilla/5.0 (iPhone; CPU iPhone OS 16_1_1 like...  Online Medical Journal   

  CHANNELTYPE                                                URL  \
0     Website  https://www.cancertherapyadvisor.com/home/canc...   
1     Website  https://www.cancertherapyadvisor.com/home/deci...   
2     Website  https://www.cancertherapyadvisor.com/home/canc...   
3     Website  https://globalrph.com/medcalcs/warfarin-mainte...   
4     Website  https://www.cureus.com/articles/58184-a-review...   

                                            KEYWORDS    TAXONOMY  IS_HCP  
0  General|Endocrine|False|Medicine|Surgery|Urolo...         NaN     0.0  
1  Bone Marrow|Radiography|Chronic|Oncology|Psych...         NaN     0.0  
2  General|Endocrine|False|Medicine|Surgery|Urolo...         NaN     0.0  
3  Dental|Total|Clinical|Pharmacology|Physicians|...  2084P0800X     1.0  
4  Critical Care|Emergency Medicine|General Pract...         NaN     0.0

In [5]:
train_df.shape

(113937, 14)

In [6]:
#SAMPLE_df.head()

In [7]:
train_df.dtypes

ID                   int64
DEVICETYPE          object
PLATFORM_ID          int64
BIDREQUESTIP        object
USERPLATFORMUID     object
USERCITY            object
USERZIPCODE        float64
USERAGENT           object
PLATFORMTYPE        object
CHANNELTYPE         object
URL                 object
KEYWORDS            object
TAXONOMY            object
IS_HCP             float64
dtype: object

In [8]:
value_counts = train_df['PLATFORMTYPE'].value_counts()
print(value_counts)

Online Medical Journal      110120
Online Learning Portal        2978
Medical News Platform          798
Electronic Health Record        40
Tele-medicine Platform           1
Name: PLATFORMTYPE, dtype: int64


In [9]:
columns_to_drop = ["ID", "DEVICETYPE", "PLATFORM_ID", "CHANNELTYPE", "USERAGENT"]
train_df.drop(columns=columns_to_drop, inplace=True)

In [10]:
for column in train_df.columns:
    print("++- "+column+" -++")
    print(train_df[column].value_counts())
    print('=======================')

++- BIDREQUESTIP -++
71.105.120.171     1718
142.154.205.202    1143
68.132.98.211       866
108.175.114.100     727
108.175.114.101     707
                   ... 
66.66.9.45            1
97.80.251.36          1
136.32.32.117         1
64.127.133.6          1
108.41.233.175        1
Name: BIDREQUESTIP, Length: 33664, dtype: int64
++- USERPLATFORMUID -++
3e2578c8-f794-41af-a38c-c5cfb3c0f014    1447
d76f7c0b-1a64-4d4b-b36d-86c832be8837    1406
d00f28d6-3a50-46cd-92f3-8601bd57ad0e     675
fcd3d327-0ad5-425e-8ea8-33ffe6a31543     629
fe53f32c-4646-4abe-afea-90b21d6b8531     617
                                        ... 
6759cbf8-cf4f-4d0e-9eb8-00808e69e841       1
bd2007ca-d0b5-4620-8078-b7c6db62037c       1
0534212a-e0e9-49e7-ace9-608e0d0eae8a       1
26a06716-2716-4ac2-9131-fa0f01f84526       1
d26887c5-15f9-4b72-80dc-728039529e60       1
Name: USERPLATFORMUID, Length: 53780, dtype: int64
++- USERCITY -++
New York          4943
Brooklyn          3237
St Louis          2545
Los Angeles

Since, multiple URLs are repeating we can use them directly to train the model. Altough breaking them up will be better for accuracy, but nonetheless we can atleast use it meaningfully.

In [11]:
null_sum = train_df.isnull().sum()

print("Sum of Null Values:")
print(null_sum)
print()

Sum of Null Values:
BIDREQUESTIP           0
USERPLATFORMUID        4
USERCITY            6359
USERZIPCODE         4592
PLATFORMTYPE           0
URL                    0
KEYWORDS               0
TAXONOMY           81624
IS_HCP                 1
dtype: int64



## Dropping the missing values in the USERPLATFORMUID as no valid relationship can be established.

In [12]:
missing_uid_rows = train_df[train_df['USERPLATFORMUID'].isnull()]
bidrequestip_values = missing_uid_rows['BIDREQUESTIP']
print(bidrequestip_values)

59158     152.11.230.114
80923      73.59.230.146
99416     152.11.230.114
102512     73.59.230.146
Name: BIDREQUESTIP, dtype: object


In [13]:
train_df.dropna(subset=['USERPLATFORMUID'], inplace=True)

## for missing value in IS_HCP

In [14]:
missing_uid_rows = train_df[train_df['IS_HCP'].isnull()]
bidrequestip_values = missing_uid_rows['BIDREQUESTIP']
print(bidrequestip_values)

73790    47.147.27.131
Name: BIDREQUESTIP, dtype: object


the missing value in IS_HCP has this BIDREQIP, so decided to follow the other values at this BIDREQIP

In [15]:
train_df['IS_HCP'] = train_df.groupby('USERPLATFORMUID')['IS_HCP'].ffill()

In [16]:
filtered_df = train_df[train_df['BIDREQUESTIP'] == '47.147.27.131']
print(filtered_df[['IS_HCP', 'BIDREQUESTIP']])

       IS_HCP   BIDREQUESTIP
412       1.0  47.147.27.131
4376      1.0  47.147.27.131
17491     1.0  47.147.27.131
73790     1.0  47.147.27.131


Since IS_HCP is binary -->!float -->>>int

In [17]:
import numpy as np
train_df['IS_HCP'] = train_df['IS_HCP'].apply(np.int64)

We can see the value of 1.0 is inserted into 73790     1.0  47.147.27.131

In [18]:
null_sum = train_df.isnull().sum()

print("Sum of Null Values:")
print(null_sum)
print()

Sum of Null Values:
BIDREQUESTIP           0
USERPLATFORMUID        0
USERCITY            6359
USERZIPCODE         4592
PLATFORMTYPE           0
URL                    0
KEYWORDS               0
TAXONOMY           81624
IS_HCP                 0
dtype: int64



In [19]:
zipcode = 482012153
usercity_values = train_df[train_df['USERZIPCODE'] == zipcode]['USERCITY']
print(usercity_values)

5009      Plymouth
12542      Detroit
19351     Plymouth
21258      Detroit
29326      Detroit
42271     Plymouth
64092     Plymouth
65229     Plymouth
66293      Detroit
70688      Detroit
87318     Plymouth
94143      Detroit
94208     Plymouth
112735         NaN
Name: USERCITY, dtype: object


In [20]:
train_df['USERCITY'] = train_df['USERCITY'].fillna(train_df.groupby('USERZIPCODE')['USERCITY'].transform('first'))

In [21]:
# null_sum = train_df.isnull().sum()

# print("Sum of Null Values:")
# print(null_sum)
# print()

In [22]:
# zipcode = 482012153
# usercity_values = train_df[train_df['USERZIPCODE'] == zipcode]['USERCITY']
# print(usercity_values)

For Taxanomy

## handling category

In [23]:
#getting all the words to list
all_keywords = train_df['KEYWORDS'].str.cat(sep='|')
word_list = list(set(all_keywords.split('|')))

# Remove empty value from word_list
word_list = list(filter(None, word_list))


list_size = len(word_list)
print("Size of word_list:", list_size)
word_list

Size of word_list: 1613


['gastrointestinal diseases',
 'opportunistic infections',
 'hamartoma',
 'carcinoma',
 'large intestine',
 'ustekinumab',
 'famciclovir',
 'cryptococcosis',
 'muscle pain',
 'FIBRINOGEN',
 'nephrology',
 'Analgesia',
 'glucagon secretion',
 'cardiology',
 'lipid profile',
 'Follicular Lymphoma',
 'muscle relaxation',
 'lymphatic diseases',
 'Minimally Invasive Surgeries',
 'lupus anticoagulant',
 'hiv infections',
 'arteriovenous malformations',
 'aphonia',
 'Protein S Deficiency',
 'transplantation',
 'yaws',
 'Preschool',
 'Cardiomyopathy',
 'Allergy Medications',
 'Lung',
 'Premature',
 'af',
 'surgical',
 'Skeleton',
 'new',
 'Premedication',
 'salivary glands',
 'anus',
 'epidural abscess',
 'Behavior',
 'encephalopathy',
 'mepolizumab',
 'Biliary Fistula',
 'catheterization',
 'dna sequencing',
 'Therapeutic Uses',
 'infertility',
 'General Hospitals',
 'janus kinase 1',
 'updated',
 'myocardial ischemia',
 'insomnia',
 'Hematology',
 'apixaban',
 'rubella',
 'staphylococcal sca

In [24]:
# # Iterate over each word in word_list
# for word in word_list:
#     # Create a new column with the word as the column name and initialize it with 0
#     train_df[word] = 0

In [25]:
# Create a new column for each word in KEYWORDS and initialize it with 0
for word in word_list:
    train_df[word] = train_df['KEYWORDS'].apply(lambda x: 1 if word in x else 0)

In [26]:
train_df.head(10)

BIDREQUESTIP                       USERPLATFORMUID       USERCITY  \
0     170.173.0.22  6974dcaa-f932-480e-9fb5-c52e20e1393a       Portland   
1    65.216.253.25  c12f3f8f-8fcf-484a-90e1-1ac04db8cdcf      Arlington   
2     66.232.79.22  a698de4b-e200-46dd-b5fb-40402175ae18    New Meadows   
3   137.54.125.246  45967533-75c8-4fbd-a00c-e6ff20447aaa           None   
4   174.202.231.99  a17e25be-532d-4cf5-b916-9308c8c3961f        Houston   
5    24.59.163.238  3b644d8f-b225-4ea3-b370-160ff7858749       Syracuse   
6     67.203.36.94  456631b8-ca59-4e7a-96a0-53bcbf85eaef    Los Angeles   
7     24.28.40.219  1e84b963-2a2a-4e1a-a767-33867aa89bd4         Dallas   
8  104.172.116.202  be61a8f7-0f00-487f-8a56-679c819fa01a  Granada Hills   
9   170.178.236.25  ddbea19a-fa49-4d3d-965a-bee29cbe92da      Davenport   

   USERZIPCODE            PLATFORMTYPE  \
0      97206.0  Online Medical Journal   
1      22202.0  Online Medical Journal   
2      83654.0  Online Medical Journal   
3  229114624.0  Online Medical Journal   
4      77008.0  Online Medical Journal   
5      13215.0  Online Medical Journal   
6      90060.0  Online Medical Journal   
7      75204.0  Online Medical Journal   
8      91344.0  Online Medical Journal   
9      52807.0  Online Medical Journal   

                                                 URL  \
0  https://www.cancertherapyadvisor.com/home/canc...   
1  https://www.cancertherapyadvisor.com/home/deci...   
2  https://www.cancertherapyadvisor.com/home/canc...   
3  https://globalrph.com/medcalcs/warfarin-mainte...   
4  https://www.cureus.com/articles/58184-a-review...   
5  https://www.cureus.com/articles/131022-the-imp...   
6           https://www.gastroenterologyadvisor.com/   
7  https://www.clinicaladvisor.com/home/features/...   
8  https://www.cancertherapyadvisor.com/home/deci...   
9   https://fpnotebook.com/Endo/Pharm/SgltInhbtr.htm   

                                            KEYWORDS    TAXONOMY  IS_HCP  \
0  General|Endocrine|False|Medicine|Surgery|Urolo...         NaN       0   
1  Bone Marrow|Radiography|Chronic|Oncology|Psych...         NaN       0   
2  General|Endocrine|False|Medicine|Surgery|Urolo...         NaN       0   
3  Dental|Total|Clinical|Pharmacology|Physicians|...  2084P0800X       1   
4  Critical Care|Emergency Medicine|General Pract...         NaN       0   
5  small|covid|nephrology|radiology|preventive me...         NaN       0   
6  Transplantation|Technology|Chronic|Neoplasms|U...         NaN       0   
7  Family Practice|Drainage|Clinical|Dermatology|...         NaN       0   
8  Intensive Care|Shock|Ischemic Heart Disease|He...  2084P0804X       1   
9  chronic kidney disease|nephrology|radiology|da...         NaN       0   

   gastrointestinal diseases  ...  pleurisy  pulmonary medicine  \
0                          0  ...         0                   0   
1                          0  ...         0                   0   
2                          0  ...         0                   0   
3                          0  ...         0                   0   
4                          0  ...         0                   0   
5                          0  ...         0                   0   
6                          0  ...         0                   0   
7                          0  ...         0                   0   
8                          0  ...         0                   0   
9                          0  ...         0                   0   

   Cardiac Output  earache  Cysts  asthma  Embolism  Conscious Sedation  \
0               0        0      0       0         0                   0   
1               0        0      0       0         0                   0   
2               0        0      0       0         0                   0   
3               0        0      0       0         0                   0   
4               0        0      0       0         0                   0   
5               0        0      0       0         0                   0   


In [27]:
#test
train_df = train_df.drop('TAXONOMY', axis=1)

In [28]:
# train_df.to_csv('train_df_with_KEYWORDS_cols.csv', index=False)

# TAXANOMY

In [29]:
train_df.head()

BIDREQUESTIP                       USERPLATFORMUID     USERCITY  \
0    170.173.0.22  6974dcaa-f932-480e-9fb5-c52e20e1393a     Portland   
1   65.216.253.25  c12f3f8f-8fcf-484a-90e1-1ac04db8cdcf    Arlington   
2    66.232.79.22  a698de4b-e200-46dd-b5fb-40402175ae18  New Meadows   
3  137.54.125.246  45967533-75c8-4fbd-a00c-e6ff20447aaa         None   
4  174.202.231.99  a17e25be-532d-4cf5-b916-9308c8c3961f      Houston   

   USERZIPCODE            PLATFORMTYPE  \
0      97206.0  Online Medical Journal   
1      22202.0  Online Medical Journal   
2      83654.0  Online Medical Journal   
3  229114624.0  Online Medical Journal   
4      77008.0  Online Medical Journal   

                                                 URL  \
0  https://www.cancertherapyadvisor.com/home/canc...   
1  https://www.cancertherapyadvisor.com/home/deci...   
2  https://www.cancertherapyadvisor.com/home/canc...   
3  https://globalrph.com/medcalcs/warfarin-mainte...   
4  https://www.cureus.com/articles/58184-a-review...   

                                            KEYWORDS  IS_HCP  \
0  General|Endocrine|False|Medicine|Surgery|Urolo...       0   
1  Bone Marrow|Radiography|Chronic|Oncology|Psych...       0   
2  General|Endocrine|False|Medicine|Surgery|Urolo...       0   
3  Dental|Total|Clinical|Pharmacology|Physicians|...       1   
4  Critical Care|Emergency Medicine|General Pract...       0   

   gastrointestinal diseases  opportunistic infections  ...  pleurisy  \
0                          0                         0  ...         0   
1                          0                         0  ...         0   
2                          0                         0  ...         0   
3                          0                         0  ...         0   
4                          0                         0  ...         0   

   pulmonary medicine  Cardiac Output  earache  Cysts  asthma  Embolism  \
0                   0               0        0      0       0         0   
1                   0               0        0      0       0         0   
2                   0               0        0      0       0         0   
3                   0               0        0      0       0         0   
4                   0               0        0      0       0         0   

   Conscious Sedation  myeloproliferative neoplasms  q fever  
0                   0                             0        0  
1                   0                             0        0  
2                   0                             0        0  
3                   0                             0        0  
4                   0                             0        0  

[5 rows x 1621 columns]

In [30]:
# from sklearn.preprocessing import LabelEncoder

# # Create an instance of LabelEncoder
# label_encoder = LabelEncoder()

# # Fit and transform the 'KEYWORDS' column
# train_df['TAXONOMY_ENCODED'] = label_encoder.fit_transform(train_df['TAXONOMY'])

# # Display the encoded values
# print(train_df['TAXONOMY_ENCODED'])

In [31]:
# import pandas as pd

# # Compute the correlation matrix
# correlation_matrix = train_df[['TAXONOMY_ENCODED'] + [
#     'diabetic retinopathy',
#     'Total',
#     'lung diseases',
#     'arrhythmias',
#     'glanders',
#     'fasting plasma glucose',
#     'Bradycardia',
#     'Tachycardia',
#     'herbal medicine',
#     'actinomycosis'
# ]].corr()

# # Display the correlation matrix
# print(correlation_matrix)


In [32]:
# import pandas as pd
# import seaborn as sns
# import matplotlib.pyplot as plt

# # Compute the correlation matrix
# correlation_matrix = train_df[['TAXONOMY_ENCODED'] + [
#     'diabetic retinopathy',
#     'Total',
#     'lung diseases',
#     'arrhythmias',
#     'glanders',
#     'fasting plasma glucose',
#     'Bradycardia',
#     'Tachycardia',
#     'herbal medicine',
#     'actinomycosis'
# ]].corr()

# # Plot the correlation matrix as a heatmap
# plt.figure(figsize=(10, 8))
# sns.heatmap(correlation_matrix, annot=True, cmap='coolwarm', square=True)
# plt.title('Correlation Matrix')
# plt.show()


In [33]:
train_df

BIDREQUESTIP                       USERPLATFORMUID      USERCITY  \
0         170.173.0.22  6974dcaa-f932-480e-9fb5-c52e20e1393a      Portland   
1        65.216.253.25  c12f3f8f-8fcf-484a-90e1-1ac04db8cdcf     Arlington   
2         66.232.79.22  a698de4b-e200-46dd-b5fb-40402175ae18   New Meadows   
3       137.54.125.246  45967533-75c8-4fbd-a00c-e6ff20447aaa          None   
4       174.202.231.99  a17e25be-532d-4cf5-b916-9308c8c3961f       Houston   
...                ...                                   ...           ...   
113932    68.82.97.126  051e4a06-4cb4-4b97-b0b1-d1c6a2b8e2c2  Philadelphia   
113933  104.172.11.109  3289bba5-4a50-4b61-a5cc-68596bbdc2bf      Van Nuys   
113934   174.21.94.113  9c36e90c-caa7-4323-8431-fb84f8293972       Seattle   
113935  69.253.129.131  afc27760-31f7-4f6e-aa08-90d8a96525d0    Wilmington   
113936  108.41.233.175  d26887c5-15f9-4b72-80dc-728039529e60  White Plains   

        USERZIPCODE            PLATFORMTYPE  \
0           97206.0  Online Medical Journal   
1           22202.0  Online Medical Journal   
2           83654.0  Online Medical Journal   
3       229114624.0  Online Medical Journal   
4           77008.0  Online Medical Journal   
...             ...                     ...   
113932      19120.0  Online Medical Journal   
113933  914014213.0  Online Medical Journal   
113934      98105.0  Online Medical Journal   
113935      19808.0  Online Medical Journal   
113936      10605.0  Online Medical Journal   

                                                      URL  \
0       https://www.cancertherapyadvisor.com/home/canc...   
1       https://www.cancertherapyadvisor.com/home/deci...   
2       https://www.cancertherapyadvisor.com/home/canc...   
3       https://globalrph.com/medcalcs/warfarin-mainte...   
4       https://www.cureus.com/articles/58184-a-review...   
...                                                   ...   
113932  https://www.clinicaladvisor.com/home/consultat...   
113933  https://www.clinicaladvisor.com/home/features/...   
113934  https://www.cureus.com/articles/39455-whats-th...   
113935  https://www.dermatologyadvisor.com/home/topics...   
113936                        https://www.medicalbag.com/   

                                                 KEYWORDS  IS_HCP  \
0       General|Endocrine|False|Medicine|Surgery|Urolo...       0   
1       Bone Marrow|Radiography|Chronic|Oncology|Psych...       0   
2       General|Endocrine|False|Medicine|Surgery|Urolo...       0   
3       Dental|Total|Clinical|Pharmacology|Physicians|...       1   
4       Critical Care|Emergency Medicine|General Pract...       0   
...                                                   ...     ...   
113932  Cardiovascular|Clinical|False|Male|Oncology|Ps...       1   
113933  Asthma|Arthritis|General|Clinical|Diagnosis|Rh...       1   
113934  Rehabilitation|Transplantation|Clinical|Preven...       1   
113935  pain|disease|autoimmune|multiple sclerosis|gra...       1   
113936  Genetic|Large|Autoimmune|Health|False|Abortion...       1   

        gastrointestinal diseases  opportunistic infections  ...  pleurisy  \
0                               0                         0  ...         0   
1                               0                         0  ...         0   
2                               0                         0  ...         0   
3                               0                         0  ...         0   
4                               0                         0  ...         0   
...                           ...                       ...  ...       ...   
113932                          0                         0  ...         0   
113933                          0                         0  ...         0   
113934                          0                         0  ...         0   
113935                          0                         0  ...         0   
113936                          0                         0  ...         0   

        p

In [34]:
import pandas as pd
from sklearn.preprocessing import OneHotEncoder


encoder = OneHotEncoder(sparse=False)
platform_type = train_df['PLATFORMTYPE'].values.reshape(-1, 1)
encoded_features = encoder.fit_transform(platform_type)
encoded_df = pd.DataFrame(encoded_features, columns=encoder.categories_[0])


df_encoded = pd.concat([train_df, encoded_df], axis=1)


print(df_encoded)
train_df = df_encoded

          BIDREQUESTIP                       USERPLATFORMUID      USERCITY  \
0         170.173.0.22  6974dcaa-f932-480e-9fb5-c52e20e1393a      Portland   
1        65.216.253.25  c12f3f8f-8fcf-484a-90e1-1ac04db8cdcf     Arlington   
2         66.232.79.22  a698de4b-e200-46dd-b5fb-40402175ae18   New Meadows   
3       137.54.125.246  45967533-75c8-4fbd-a00c-e6ff20447aaa          None   
4       174.202.231.99  a17e25be-532d-4cf5-b916-9308c8c3961f       Houston   
...                ...                                   ...           ...   
113936  108.41.233.175  d26887c5-15f9-4b72-80dc-728039529e60  White Plains   
59158              NaN                                   NaN           NaN   
80923              NaN                                   NaN           NaN   
99416              NaN                                   NaN           NaN   
102512             NaN                                   NaN           NaN   

        USERZIPCODE            PLATFORMTYPE  \
0           9720

In [35]:
train_df.drop('PLATFORMTYPE', axis=1, inplace=True)

In [36]:
train_df.shape

(113937, 1625)

In [37]:
train_df.columns

Index(['BIDREQUESTIP', 'USERPLATFORMUID', 'USERCITY', 'USERZIPCODE', 'URL',
       'KEYWORDS', 'IS_HCP', 'gastrointestinal diseases',
       'opportunistic infections', 'hamartoma',
       ...
       'asthma', 'Embolism', 'Conscious Sedation',
       'myeloproliferative neoplasms', 'q fever', 'Electronic Health Record',
       'Medical News Platform', 'Online Learning Portal',
       'Online Medical Journal', 'Tele-medicine Platform'],
      dtype='object', length=1625)

In [38]:
train_df['USERCITY'].isna().sum()

5929

In [39]:
train_df['USERZIPCODE'].isna().sum()

4596

In [40]:
train_df.dropna(subset=['USERCITY'], inplace=True)

In [41]:
train_df.shape

(108008, 1625)

In [42]:
train_df.dropna(subset=['USERZIPCODE'], inplace=True)

In [43]:
train_df.shape

(108006, 1625)

In [44]:
columns_to_drop = ['BIDREQUESTIP', 'USERPLATFORMUID','URL','KEYWORDS']
train_df.drop(columns_to_drop, axis=1, inplace=True)

In [45]:
train_df.head(5)

USERCITY  USERZIPCODE  IS_HCP  gastrointestinal diseases  \
0     Portland      97206.0     0.0                        0.0   
1    Arlington      22202.0     0.0                        0.0   
2  New Meadows      83654.0     0.0                        0.0   
4      Houston      77008.0     0.0                        0.0   
5     Syracuse      13215.0     0.0                        0.0   

   opportunistic infections  hamartoma  carcinoma  large intestine  \
0                       0.0        0.0        0.0              0.0   
1                       0.0        0.0        0.0              0.0   
2                       0.0        0.0        0.0              0.0   
4                       0.0        0.0        0.0              0.0   
5                       0.0        0.0        0.0              0.0   

   ustekinumab  famciclovir  ...  asthma  Embolism  Conscious Sedation  \
0          0.0          0.0  ...     0.0       0.0                 0.0   
1          0.0          0.0  ...     0.0       0.0                 0.0   
2          0.0          0.0  ...     0.0       0.0                 0.0   
4          0.0          0.0  ...     0.0       0.0                 0.0   
5          0.0          0.0  ...     0.0       0.0                 0.0   

   myeloproliferative neoplasms  q fever  Electronic Health Record  \
0                           0.0      0.0                       0.0   
1                           0.0      0.0                       0.0   
2                           0.0      0.0                       0.0   
4                           0.0      0.0                       0.0   
5                           0.0      0.0                       0.0   

   Medical News Platform  Online Learning Portal  Online Medical Journal  \
0                    0.0                     0.0                     1.0   
1                    0.0                     0.0                     1.0   
2                    0.0                     0.0                     1.0   
4                    0.0                     0.0                     1.0   
5                    0.0                     0.0                     1.0   

   Tele-medicine Platform  
0                     0.0  
1                     0.0  
2                     0.0  
4                     0.0  
5                     0.0  

[5 rows x 1621 columns]

In [46]:
import pandas as pd
from sklearn.preprocessing import LabelEncoder

city_encoder = LabelEncoder()
train_df['USERCITY'] = city_encoder.fit_transform(train_df['USERCITY'])


zipcode_encoder = LabelEncoder()
train_df['USERZIPCODE'] = zipcode_encoder.fit_transform(train_df['USERZIPCODE'])


train_df.head(5)


USERCITY  USERZIPCODE  IS_HCP  gastrointestinal diseases  \
0      3176         8934     0.0                        0.0   
1       130         2207     0.0                        0.0   
2      2700         7585     0.0                        0.0   
4      1766         7018     0.0                        0.0   
5      3877         1351     0.0                        0.0   

   opportunistic infections  hamartoma  carcinoma  large intestine  \
0                       0.0        0.0        0.0              0.0   
1                       0.0        0.0        0.0              0.0   
2                       0.0        0.0        0.0              0.0   
4                       0.0        0.0        0.0              0.0   
5                       0.0        0.0        0.0              0.0   

   ustekinumab  famciclovir  ...  asthma  Embolism  Conscious Sedation  \
0          0.0          0.0  ...     0.0       0.0                 0.0   
1          0.0          0.0  ...     0.0       0.0                 0.0   
2          0.0          0.0  ...     0.0       0.0                 0.0   
4          0.0          0.0  ...     0.0       0.0                 0.0   
5          0.0          0.0  ...     0.0       0.0                 0.0   

   myeloproliferative neoplasms  q fever  Electronic Health Record  \
0                           0.0      0.0                       0.0   
1                           0.0      0.0                       0.0   
2                           0.0      0.0                       0.0   
4                           0.0      0.0                       0.0   
5                           0.0      0.0                       0.0   

   Medical News Platform  Online Learning Portal  Online Medical Journal  \
0                    0.0                     0.0                     1.0   
1                    0.0                     0.0                     1.0   
2                    0.0                     0.0                     1.0   
4                    0.0                     0.0                     1.0   
5                    0.0                     0.0                     1.0   

   Tele-medicine Platform  
0                     0.0  
1                     0.0  
2                     0.0  
4                     0.0  
5                     0.0  

[5 rows x 1621 columns]

In [47]:
# X_train = feature_engineered_train_data.drop('HCP', axis=1)
# y_train = feature_engineered_train_data['HCP']

# # Split the test dataset into features and target variable
# X_test = feature_engineered_test_data.drop('HCP', axis=1)
# y_test = feature_engineered_test_data['HCP']

# # Define the XGBoost classifier
# model = xgb.XGBClassifier()

# # Train the model
# model.fit(X_train, y_train)

# # Make predictions on the test set
# y_pred = model.predict(X_test)

# # Evaluate the model
# accuracy = accuracy_score(y_test, y_pred)
# print("Accuracy: {:.2f}%".format(accuracy * 100))

Changes made for the test_df file in the below section:

In [48]:
test_df.head(5)

ID DEVICETYPE  PLATFORM_ID     BIDREQUESTIP  \
0  115501    Desktop            2   75.189.231.103   
1  115502     Mobile            2    24.101.33.158   
2  115503    Desktop            2  172.118.216.142   
3  115504    Desktop            7   71.105.120.171   
4  115505    Desktop            2     73.82.211.73   

                        USERPLATFORMUID       USERCITY  USERZIPCODE  \
0  0d5041ff-f0b6-4d1a-9ad7-0a29f7d485b4   Fayetteville      28305.0   
1  c8396dd0-969f-4d99-a40b-b7bb1f516154  Conneaut Lake      16316.0   
2  3c97a081-6518-43f8-9f26-369759cfb471         Covina      91724.0   
3  3e2578c8-f794-41af-a38c-c5cfb3c0f014       Brooklyn      11226.0   
4  ec2ae7ce-6a8c-4156-98a7-07203e60f483       Marietta      30062.0   

                                           USERAGENT            PLATFORMTYPE  \
0  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6...  Online Medical Journal   
1  Mozilla/5.0 (iPhone; CPU iPhone OS 15_6_1 like...  Online Medical Journal   
2  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7...  Online Medical Journal   
3  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7...  Online Medical Journal   
4  Mozilla/5.0 (Windows NT 10.0; Win64; x64) Appl...  Online Medical Journal   

  CHANNELTYPE                                                URL  \
0     Website  https://www.clinicaladvisor.com/home/features/...   
1     Website  https://www.ophthalmologyadvisor.com/topics/ca...   
2     Website  https://www.psychiatryadvisor.com/author/tori-...   
3     Website  https://www.cureus.com/articles/105482-diverti...   
4     Website  https://www.renalandurologynews.com/home/confe...   

                                            KEYWORDS  
0  Family Practice|Drainage|Clinical|Dermatology|...  
1  General|Clinical|Operative|Medicine|Cardiology...  
2  Abortion|Anxiety Disorders|Apnea|False|Trauma|...  
3  Health|Male|Neurological Surgery|Otolaryngolog...  
4  chronic kidney disease|pain|nephrology|disease...

In [49]:
import pandas as pd
import numpy as np
from sklearn.preprocessing import OneHotEncoder, LabelEncoder


columns_to_drop = ['ID', 'DEVICETYPE', 'PLATFORM_ID', 'CHANNELTYPE', 'USERAGENT',
                   'BIDREQUESTIP', 'USERPLATFORMUID', 'URL']


test_df.drop(columns=columns_to_drop, inplace=True)

In [50]:
test_df['USERCITY'] = test_df['USERCITY'].fillna(test_df.groupby('USERZIPCODE')['USERCITY'].transform('first'))


all_keywords = test_df['KEYWORDS'].str.cat(sep='|')
word_list = list(set(all_keywords.split('|')))


word_list = list(filter(None, word_list))


for word in word_list:
    test_df[word] = test_df['KEYWORDS'].apply(lambda x: 1 if word in x else 0)


columns_to_drop = ['KEYWORDS']
test_df.drop(columns_to_drop, axis=1, inplace=True)


encoder = OneHotEncoder(sparse=False)
platform_type = test_df['PLATFORMTYPE'].values.reshape(-1, 1)
encoded_features = encoder.fit_transform(platform_type)
encoded_df = pd.DataFrame(encoded_features, columns=encoder.categories_[0])


test_df = pd.concat([test_df, encoded_df], axis=1)


test_df.drop('PLATFORMTYPE', axis=1, inplace=True)


test_df.dropna(subset=['USERCITY', 'USERZIPCODE'], inplace=True)


city_encoder = LabelEncoder()
test_df['USERCITY'] = city_encoder.fit_transform(test_df['USERCITY'])

zipcode_encoder = LabelEncoder()
test_df['USERZIPCODE'] = zipcode_encoder.fit_transform(test_df['USERZIPCODE'])

# Print the updated DataFrame
print(test_df.head(5))


   USERCITY  USERZIPCODE  gastrointestinal diseases  opportunistic infections  \
0       857         1827                          0                         0   
1       559         1091                          0                         0   
2       592         5481                          0                         0   
3       323          739                          0                         0   
4      1609         2008                          0                         0   

   hamartoma  carcinoma  large intestine  ustekinumab  famciclovir  \
0          0          0                0            0            0   
1          0          0                0            0            0   
2          0          0                0            0            0   
3          0          0                0            0            0   
4          0          1                0            0            0   

   cryptococcosis  ...  Cardiac Output  Cysts  asthma  Embolism  \
0               0  ...   

In [51]:
test_df

USERCITY  USERZIPCODE  gastrointestinal diseases  \
0           857         1827                          0   
1           559         1091                          0   
2           592         5481                          0   
3           323          739                          0   
4          1609         2008                          0   
...         ...          ...                        ...   
28487      2436         5898                          0   
28488       323          730                          0   
28489      1693         2340                          0   
28490       903         2355                          0   
28492      2987         1844                          0   

       opportunistic infections  hamartoma  carcinoma  large intestine  \
0                             0          0          0                0   
1                             0          0          0                0   
2                             0          0          0                0   
3                             0          0          0                0   
4                             0          0          1                0   
...                         ...        ...        ...              ...   
28487                         0          0          0                0   
28488                         0          0          0                0   
28489                         0          0          0                0   
28490                         0          0          0                0   
28492                         0          0          0                0   

       ustekinumab  famciclovir  cryptococcosis  ...  Cardiac Output  Cysts  \
0                0            0               0  ...               0      0   
1                0            0               0  ...               0      0   
2                0            0               0  ...               0      0   
3                0            0               0  ...               0      0   
4                0            0               0  ...               0      0   
...            ...          ...             ...  ...             ...    ...   
28487            0            0               0  ...               0      0   
28488            0            0               0  ...               0      0   
28489            0            0               0  ...               0      0   
28490            0            0               0  ...               0      0   
28492            0            0               0  ...               0      0   

       asthma  Embolism  Conscious Sedation  myeloproliferative neoplasms  \
0           0         0                   0                             0   
1           0         0                   0                             0   
2           0         0                   0                             0   
3           0         0                   0                             0   
4           0         0                   0                             0   
...       ...       ...                 ...                           ...   
28487       0         0                   0                             0   
28488       0         0                   0                             0   
28489       0         0                   0                             0   
28490       0         0                   0                             0   
28492       0         0                   0                             0   

       Electronic Health Record  Medical News Platform  \
0                           0.0                    0.0   
1                           0.0                    0.0   
2                           0.0                    0.0   
3                           0.0                    0.0   
4                           0.0                    0.0   
...                         ...                    ...   
28487                       0.0                    0.0   
28488                       0.0                    0.0   
28489        

In [52]:
test_df

USERCITY  USERZIPCODE  gastrointestinal diseases  \
0           857         1827                          0   
1           559         1091                          0   
2           592         5481                          0   
3           323          739                          0   
4          1609         2008                          0   
...         ...          ...                        ...   
28487      2436         5898                          0   
28488       323          730                          0   
28489      1693         2340                          0   
28490       903         2355                          0   
28492      2987         1844                          0   

       opportunistic infections  hamartoma  carcinoma  large intestine  \
0                             0          0          0                0   
1                             0          0          0                0   
2                             0          0          0                0   
3                             0          0          0                0   
4                             0          0          1                0   
...                         ...        ...        ...              ...   
28487                         0          0          0                0   
28488                         0          0          0                0   
28489                         0          0          0                0   
28490                         0          0          0                0   
28492                         0          0          0                0   

       ustekinumab  famciclovir  cryptococcosis  ...  Cardiac Output  Cysts  \
0                0            0               0  ...               0      0   
1                0            0               0  ...               0      0   
2                0            0               0  ...               0      0   
3                0            0               0  ...               0      0   
4                0            0               0  ...               0      0   
...            ...          ...             ...  ...             ...    ...   
28487            0            0               0  ...               0      0   
28488            0            0               0  ...               0      0   
28489            0            0               0  ...               0      0   
28490            0            0               0  ...               0      0   
28492            0            0               0  ...               0      0   

       asthma  Embolism  Conscious Sedation  myeloproliferative neoplasms  \
0           0         0                   0                             0   
1           0         0                   0                             0   
2           0         0                   0                             0   
3           0         0                   0                             0   
4           0         0                   0                             0   
...       ...       ...                 ...                           ...   
28487       0         0                   0                             0   
28488       0         0                   0                             0   
28489       0         0                   0                             0   
28490       0         0                   0                             0   
28492       0         0                   0                             0   

       Electronic Health Record  Medical News Platform  \
0                           0.0                    0.0   
1                           0.0                    0.0   
2                           0.0                    0.0   
3                           0.0                    0.0   
4                           0.0                    0.0   
...                         ...                    ...   
28487                       0.0                    0.0   
28488                       0.0                    0.0   
28489        

In [53]:
train_df

USERCITY  USERZIPCODE  IS_HCP  gastrointestinal diseases  \
0           3176         8934     0.0                        0.0   
1            130         2207     0.0                        0.0   
2           2700         7585     0.0                        0.0   
4           1766         7018     0.0                        0.0   
5           3877         1351     0.0                        0.0   
...          ...          ...     ...                        ...   
113932      3066         1837     1.0                        0.0   
113933      4042        10318     1.0                        0.0   
113934      3569         9052     1.0                        0.0   
113935      4321         1932     1.0                        0.0   
113936      4278          994     1.0                        0.0   

        opportunistic infections  hamartoma  carcinoma  large intestine  \
0                            0.0        0.0        0.0              0.0   
1                            0.0        0.0        0.0              0.0   
2                            0.0        0.0        0.0              0.0   
4                            0.0        0.0        0.0              0.0   
5                            0.0        0.0        0.0              0.0   
...                          ...        ...        ...              ...   
113932                       0.0        0.0        0.0              0.0   
113933                       0.0        0.0        0.0              0.0   
113934                       0.0        0.0        0.0              0.0   
113935                       0.0        0.0        0.0              0.0   
113936                       0.0        0.0        0.0              0.0   

        ustekinumab  famciclovir  ...  asthma  Embolism  Conscious Sedation  \
0               0.0          0.0  ...     0.0       0.0                 0.0   
1               0.0          0.0  ...     0.0       0.0                 0.0   
2               0.0          0.0  ...     0.0       0.0                 0.0   
4               0.0          0.0  ...     0.0       0.0                 0.0   
5               0.0          0.0  ...     0.0       0.0                 0.0   
...             ...          ...  ...     ...       ...                 ...   
113932          0.0          0.0  ...     0.0       0.0                 0.0   
113933          0.0          0.0  ...     0.0       0.0                 0.0   
113934          0.0          0.0  ...     0.0       0.0                 0.0   
113935          0.0          0.0  ...     0.0       0.0                 0.0   
113936          0.0          0.0  ...     0.0       0.0                 0.0   

        myeloproliferative neoplasms  q fever  Electronic Health Record  \
0                                0.0      0.0                       0.0   
1                                0.0      0.0                       0.0   
2                                0.0      0.0                       0.0   
4                                0.0      0.0                       0.0   
5                                0.0      0.0                       0.0   
...                              ...      ...                       ...   
113932                           0.0      0.0                       0.0   
113933                           0.0      0.0                       NaN   
113934                           0.0      0.0                       NaN   
113935                           0.0      0.0                       NaN   
113936                           0.0      0.0                       NaN   

        Medical News Platform  Online Learning Portal  Online Medical Journal  \
0                         0.0                     0.0                     1.0   
1                         0.0                     0.0                     1.0   
2                         0.0                     0.0                     1.0   
4                         0.0                     0.0                     1.0   
5                         0.0      

# Inserting the missing columns in the test_df fron the train_df 

In [71]:
train_columns = set(train_df.columns)
test_columns = set(test_df.columns)

missing_columns = train_columns - test_columns

for column in missing_columns:
    test_df[column] = 0

In [68]:
train_df.shape

(108006, 1627)

In [69]:
test_df.shape

(27008, 1627)

In [70]:
train_columns = set(train_df.columns)
test_columns = set(test_df.columns)

missing_columns = test_columns - train_columns

for column in missing_columns:
    train_df[column] = 0
    # print(column)

In [63]:
train_columns = set(train_df.columns)
test_columns = set(test_df.columns)

missing_columns = train_columns - test_columns

for column in missing_columns:
    # train_df[column] = 0
    print(column)

neutron capture therapy
dpp-4
thromboplastin
nasal bone
retinal vein occlusion
Tele-medicine Platform
mycoses
chagas disease
Imperforate
Axotomy
tracheomalacia
cystitis
bell palsy
blood coagulation disorders
impulse control
blood protein disorders
adrenal hyperplasia
elephantiasis
empyema
Decompressive Craniectomy
iliotibial band syndrome
hemostatic disorders
Chemoprevention
intestinal mucosa
cervical rib syndrome
mediastinitis
general hospitals
Minimally Invasive Surgical Procedures
nasal septum
Juvenile Dermatomyositis
rocky mountain spotted fever
Follicular Lymphoma
palatine tonsil
muscle relaxation
latent tuberculosis
blunt cardiac injury
tonsillitis
persistent infection
porphyrias
vitamin k deficiency
diclofenac
erysipelas
aphonia
Thoracic Surgical Procedures
filariasis
glanders
behcet syndrome
biliary atresia
parasitic diseases
burkitt lymphoma
anthropometry
tabes dorsalis
health education
bone marrow diseases
hepatic veno-occlusive disease
Medical Specialties
coinfection
varicos

In [73]:
test_df['IS_HCP'].eq(0).sum()

27008

In [75]:
train_column_order = train_df.columns

test_df = test_df.reindex(columns=train_column_order)

In [58]:
# Drop the "IS_HCP" column from test_df
# test_df = test_df.drop("IS_HCP", axis=1)

KeyError: "['IS_HCP'] not found in axis"

In [60]:
import pandas as pd
import xgboost as xgb
from sklearn.metrics import mean_squared_error
from math import sqrt


X_train = train_df.drop('IS_HCP', axis=1)
y_train = train_df['IS_HCP']


X_test = test_df.drop('IS_HCP', axis=1)

y_test = test_df['IS_HCP']


model = xgb.XGBRegressor()


model.fit(X_train, y_train)


y_pred = model.predict(X_test)


mse = mean_squared_error(y_test, y_pred)
print("Mean Squared Error (MSE): {:.2f}".format(mse))


rmse = sqrt(mse)
print("Root Mean Squared Error (RMSE): {:.2f}".format(rmse))

KeyError: 'IS_HCP'

In [78]:
import pandas as pd
import xgboost as xgb
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score



X = train_df.drop('IS_HCP', axis=1)
y = train_df['IS_HCP']

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=0)


model = xgb.XGBClassifier()


model.fit(X_train, y_train)


y_pred = model.predict(X_test)


accuracy = accuracy_score(y_test, y_pred)
print("Accuracy: {:.2f}%".format(accuracy * 100))

Accuracy: 97.39%


In [79]:
mse = mean_squared_error(y_test, y_pred)
print("Mean Squared Error (MSE): {:.2f}".format(mse))

rmse = sqrt(mse)
print("Root Mean Squared Error (RMSE): {:.2f}".format(rmse))

Mean Squared Error (MSE): 0.03
Root Mean Squared Error (RMSE): 0.16


In [77]:
y_pred

array([0, 1, 0, ..., 0, 0, 1])

In [80]:
import pandas as pd
import xgboost as xgb
from sklearn.model_selection import cross_val_score


X = train_df.drop('IS_HCP', axis=1)
y = train_df['IS_HCP']


model = xgb.XGBClassifier()


scores = cross_val_score(model, X, y, cv=5)  
accuracy = scores.mean()

print("Accuracy: {:.2f}%".format(accuracy * 100))

Accuracy: 97.37%


In [82]:
model.fit(X, y)
X_test = test_df.drop('IS_HCP', axis=1)
y_pred = model.predict(X_test)

In [84]:
len(y_pred)

27008

In [ ]:
accuracy = accuracy_score(y_test, y_pred)
print("Accuracy: {:.2f}%".format(accuracy * 100))

In [ ]:

print("Accuracy: {:.2f}%".format(accuracy * 100))

Accuracy: 97.50%
